hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -1105,6 +1111,54 @@ public StartContainerBalancerResponseProto startContainerBalancer(
       cbc.setMaxSizeLeavingSource(msls * OzoneConsts.GB);
     }
 
+    if (balancingInterval.isPresent()) {
+      long bi = balancingInterval.get();
+      auditMap.put("balancingInterval", String.valueOf(bi));
+      Preconditions.checkState(bi > 0,
+              "balancingInterval must be greater than zero.");
+      cbc.setBalancingInterval(bi);
+    }
+
+    if (moveTimeout.isPresent()) {
+      long mt = moveTimeout.get();
+      auditMap.put("moveTimeout", String.valueOf(mt));
+      Preconditions.checkState(mt > 0,
+              "moveTimeout must be greater than zero.");
+      cbc.setMoveTimeout(mt);
+    }
+
+    if (moveReplicationTimeout.isPresent()) {
+      long mrt = moveReplicationTimeout.get();
+      auditMap.put("moveReplicationTimeout", String.valueOf(mrt));
+      Preconditions.checkState(mrt > 0,
+              "moveTimeout must be greater than zero.");
```

Comment:
These checks will crash SCM if the client provides incorrect values. It is better to:
- Use a [validate](https://github.com/apache/ozone/blob/2242f5ca91ac802f3d5ddaca1c9f7389012f8227/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerScannerConfiguration.java#L142) method to make sure the configs are set correctly on startup.
- Return a proper error to the client when an invalid option has been used from the CLI.

Reply:
On providing incorrect values for moveReplicationTimeout, moveTimeout and balancingInterval, it throws an error message specified in the Preconditions.checkState() and exits. It doesn't crash the SCM.
<img width="1680" alt="Screenshot 2024-02-28 at 4 54 13 PM" src="https://github.com/apache/ozone/assets/79865743/4ea8729c-8aaa-4aa7-b937-58679d950a96">

Reply:
Hmm looks like Hadoop RPC is automatically doing some conversion to get the Runtime exception message back to the client? I can't figure out if the handler thread is crashing at the end of this or not, but it's probably a daemon so SCM keeps running. Better to explicitly throw a checked exception and make sure the response proto is updated so [this code runs](https://github.com/apache/ozone/blob/886fc3d419e07a5736a2112fd748c3b212c2f11f/hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/ContainerBalancerStartSubcommand.java#L86-L90).

Reply:
We should also make the error messages more user facing. For example "Move timeout" instead of "moveTimeout".

Reply:
@errose28 good point. I wonder if we can even skip throwing a checked exception and simply audit the failure, wrap the failure message in the response proto and send it back? 

SCM client has retry and failover functionalities - the proxy will, by default (unless we explicitly code to not failover/retry for a particular exception), try to failover to a different SCM and retry the command there if it receives an `IOException`, for example. These retries are unnecessary in this case because the provided configs are incorrect. To change this behaviour we'd need to add some logic for the client to just fail and not retry or failover to another proxy. This is doable but more complicated.

How about throwing at the client side instead if the command failed, so that it returns with a non zero exit code?
```
    if (response.getStart()) {
      System.out.println("Container Balancer started successfully.");
    } else {
      System.out.println("Failed to start Container Balancer.");
      if (response.hasMessage()) {
        System.out.printf("Failure reason: %s", response.getMessage());
      }
    }
```

Reply:
@sarvekshayr are you planning to make this change as well?

Reply:
No, I'll keep the code as it is because the checks are first handled by the client side in `StorageContainerLocationProtocolClientSideTranslatorPB.java`, upon failure it returns with a non zero exit status.

Reply:
Client side checks are a good optimization but the server needs to be robust in its own error detection. Additionally we should have consistent ways to handle client errors in SCM. See other methods' error handling like [close containe](https://github.com/apache/ozone/blob/573ec30710625dc234e33e340a49268a4672c791/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java#L706)r for an example of how to implement this.

Reply:
I've added SCMException to handle the checks on the server side. Please review it @errose28

Reply:
I think we can make the code simpler by just using an `IOException` instead of an `SCMException`. In this method we're creating an exception only to catch it later and audit the failure. The original pattern was probably to create an exception, audit it, and then throw it back to the caller, therefore creating an exception was required in the first place. 

Here, it's sufficient to just use an IOException since we are just catching it. Even simpler and maybe faster would be to not use exceptions at all... but we don't need to go there in this pr.

Reply:
> I think we can make the code simpler by just using an `IOException` instead of an `SCMException`. 

I've made the required changes.